### PR TITLE
Implement new hardfork for v17 to be used in future changes

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
Not sure I'm a huge fan of `DEPLOYMENT_V17` name, lmk if there are any suggestions

### CODE REVIEW

**Comment by thephez** [comment]
Will the fork primarily be to support platform requirements? If so, the name could reflect that.

File: src/consensus/params.h [hunk 1]
```diff
@@ -21,6 +21,7 @@ enum DeploymentPos
     DEPLOYMENT_DIP0003, // Deployment of DIP0002 and DIP0003 (txv3 and deterministic MN lists)
     DEPLOYMENT_DIP0008, // Deployment of ChainLock enforcement
     DEPLOYMENT_REALLOC, // Deployment of Block Reward Reallocation
+    DEPLOYMENT_V17, // Deployment of Block Reward Reallocation
```
Comment (UdjinM6) [inline src/consensus/params.h:24]:
```suggestion
    DEPLOYMENT_V17, // Deployment of V17 Hard Fork
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by UdjinM6** [comment]
Agree that a better name should be picked at some point but we can tweak this later, it's not a blocker imo.

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by xdustinface — APPROVED** [review]
utACK

**Pull request merged**